src/identities/mnemonic.ts

```diff
@@ -1,30 +1,212 @@
+import { ECPair, payments, Psbt } from 'liquidjs-lib';
 import * as bip39 from 'bip39';
+import { fromSeed as slip77fromSeed, Slip77Interface } from 'slip77';
+import { fromSeed as bip32fromSeed, BIP32Interface } from 'bip32';
 import Identity, {
   AddressInterface,
   IdentityInterface,
   IdentityOpts,
+  IdentityType,
 } from '../identity';
+import { BufferMap } from '../utils';
 
+export interface MnemonicOptsValue {
+  mnemonic: string;
+  language?: string;
+}
+
+function instanceOfMnemonicOptsValue(value: any): value is MnemonicOptsValue {
+  return 'mnemonic' in value;
+}
+
+interface AddressInterfaceExtended {
+  address: AddressInterface;
+  signingPrivateKey: string;
+  derivationPath: string;
+}
+
+/**
+ * @class Mnemonic
+ * Get a mnemonic as parameter to set up an HD Wallet.
+ * @member masterPrivateKeyNode a BIP32 node computed from the seed, used to generate signing key pairs.
+ * @member masterBlindingKeyNode a SLIP77 node computed from the seed, used to generate the blinding key pairs.
+ * @member derivationPath the base derivation path.
+ * @member index the next index used to derive the base node (for signing key pairs).
+ * @member scriptToAddressCache a map scriptPubKey --> address generation.
+ */
 export default class Mnemonic extends Identity implements IdentityInterface {
-  private words: Array<string>;
+  static INITIAL_BASE_PATH: string = "m/84'/0'/0'/0'";
```

Comment:
```suggestion
  static INITIAL_BASE_PATH: string = "m/84'/0'/0'";
```


bip44 has 5 levels: `m / purpose' / coin_type' / account' / change / address_index` 

Theoretically the base path should be only m/84'

Since we don't want to encourage users to reuse the same seed for both mainnet and regtest (but rather have two separated) we usually hardcode also coin type to 0 (he network).

Also multiple accounts here are of no use and not planned in the SDK, and we can hardcode to 0 also that (ie. using only one)

Reply:
And what about purpose ?

Reply:
purpose is 84 (ie native segwit)

Reply:
whaaaaaat :0 I had skipped this information thank u. It explains the extra '0' ^^